src/cli/commands/run.js

```diff
@@ -62,30 +62,37 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
     if (pkgScripts && action in pkgScripts) {
       const preAction = `pre${action}`;
       if (preAction in pkgScripts) {
-        cmds.push([preAction, pkgScripts[preAction]]);
+        cmds.push([preAction, pkgScripts[preAction], false]);
       }
 
-      cmds.push([action, scripts[action]]);
+      cmds.push([action, scripts[action], true]);
 
       const postAction = `post${action}`;
       if (postAction in pkgScripts) {
-        cmds.push([postAction, pkgScripts[postAction]]);
+        cmds.push([postAction, pkgScripts[postAction], false]);
       }
     } else if (scripts[action]) {
-      cmds.push([action, scripts[action]]);
+      cmds.push([action, scripts[action], true]);
     }
 
     if (cmds.length) {
       // Disable wrapper in executed commands
       process.env.YARN_WRAP_OUTPUT = 'false';
-      for (const [stage, cmd] of cmds) {
+      for (const [stage, cmd, isInteractive] of cmds) {
         // only tack on trailing arguments for default script, ignore for pre and post - #1595
         const cmdWithArgs = stage === action ? sh`${unquoted(cmd)} ${args}` : cmd;
         const customShell = config.getOption('script-shell');
         if (customShell) {
-          await execCommand(stage, config, cmdWithArgs, config.cwd, String(customShell));
+          await execCommand({
+            stage,
+            config,
+            cmd: cmdWithArgs,
+            cwd: config.cwd,
+            isInteractive,
```

Comment:
If we decide to make `postX` and `preX` scripts interactive, then this line should simply become `isInteractive: true` and all other changes above should be removed.

Reply:
I think the pre/post scripts can be left as non-interactive, this should be fine